web/ui/react-app/src/pages/targets/EndpointLink.tsx

```diff
@@ -8,23 +8,23 @@ export interface EndpointLinkProps {
 
 const EndpointLink: FC<EndpointLinkProps> = ({ endpoint, globalUrl }) => {
   let url: URL;
+  let search = '';
+  let invalidURL = false;
   try {
     url = new URL(endpoint);
   } catch (err: unknown) {
-    const error = err as Error;
-    return (
-      <Alert color="danger">
-        <strong>Error:</strong> {error.message}
-      </Alert>
-    );
+    invalidURL = true;
+    if (endpoint.indexOf('?') > -1) search = endpoint.substring(endpoint.indexOf('?'));
+    url = new URL('http://0.0.0.0' + search);
   }
 
   const { host, pathname, protocol, searchParams }: URL = url;
   const params = Array.from(searchParams.entries());
-
+  const href = invalidURL ? endpoint : globalUrl;
+  const displayLink = invalidURL ? endpoint.replace(search, '') : `${protocol}//${host}${pathname}`;
   return (
     <>
-      <a href={globalUrl}>{`${protocol}//${host}${pathname}`}</a>
+      <a href={href}>{displayLink}</a>
```

Comment:
we should not href if the URL is "invalid"

Reply:
For what it’s worth, https://github.com/prometheus/prometheus/issues/9760#issuecomment-970623861

Reply:
The URL being "invalid" only means that it can't be parsed, not that it necessarily can't be navigated to.

Reply:
changed href to globalURL